### PR TITLE
fix: status indicator refactoring to Object Status

### PR DESCRIPTION
**Pull request by Lokanathan-k**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/2137

#### Please provide a brief summary of this pull request.

introducing the new component with refactoring of the Status indicator

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
Hi @Lokanathan-k , 
The team is not reviewing PRs that have failing builds (even if it's an "internal" PR). Please fix the errors so we can start reviewing the PR. 
I still took a look at it and I have the same comments as InfoLabel. Please use the `CssClassBuilder` interface to build the classes. You can take a look at components like Message Strip (currently in review), Bar, Tabs, Button, etc. where we adopted this approach. If you have any questions please let me know. Thank you!

File: libs/core/src/lib/object-status/object-status.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    color: String;
+
+    @Input()
+    inverted: boolean;
+
+    @Input()
+    clickable: boolean;
+
+    @Input()
+    large: boolean;
```
Comment (InnaAtanasova) [inline libs/core/src/lib/object-status/object-status.component.ts]: Should be private variables with getters and setters.

File: libs/core/src/lib/object-status/object-status.component.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+        if (this.clickable) {
+            this.cssname = this.cssname + ' fd-object-status--link';
+        }
+        if (this.large) {
+            this.cssname = this.cssname + ' fd-object-status--large';
+        }
+
+        return this.cssname;
+    }
```
Comment (InnaAtanasova) [inline libs/core/src/lib/object-status/object-status.component.ts]: Should use `buildComponentCssClass()`

File: apps/docs/src/app/core/api-files.ts [hunk 3]
```diff
@@ -195,6 +195,10 @@ export const API_FILES = {
         'NotificationDefault',
         'NotificationRef',
     ],
+    objectstatus: [
```
Comment (salarenko) [inline apps/docs/src/app/core/api-files.ts]: Should be: `objectStatus`

File: apps/docs/src/app/core/component-docs/object-status/examples/object-status-clickable-and-icon-example.component.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<a fd-object-status [status]="'negative'" [glyph]="'status-negative'" [clickable]="true">Negative</a>
+<a fd-object-status [status]="'critical'" [glyph]="'status-critical'" [clickable]="true">Critical</a>
+<a fd-object-status [status]="'positive'" [glyph]="'status-positive'" [clickable]="true">Positive</a>
+<a fd-object-status [status]="'informative'" [glyph]="'hint'" [clickable]="true">Informative</a>
+<a fd-object-status [glyph]="'to-be-reviewed'" [clickable]="true">Default</a>
+<br>
+<br>
+<ng-templete *ngFor="let item of [1,2,3,4,5,6,7,8]">
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/object-status/examples/object-status-clickable-and-icon-example.component.html:8]: I think you should rather use `<ng-container>` here.

File: apps/docs/src/app/core/component-docs/object-status/examples/object-status-generic-text-example.component.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<ng-templete *ngFor="let item of [1,2,3,4,5,6,7,8]">
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/object-status/examples/object-status-generic-text-example.component.html:1]: I think you should rather use `<ng-container>` here.

File: apps/docs/src/app/core/component-docs/object-status/examples/object-status-inverted-generic-text-example.component.html [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<ng-templete *ngFor="let item of [1,2,3,4,5,6,7,8]">
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/object-status/examples/object-status-inverted-generic-text-example.component.html:1]: I think you should rather use `<ng-container>` here.

**Pull request merged**